Intersect.Client/Entities/Player.cs

```diff
@@ -2251,125 +2251,124 @@ public virtual void DrawGuildName(Color textColor, Color borderColor = null, Col
 
-                                    // If NpcOverheadInfo is toggled off, draw NPCs
-                                    // overhead information only when hovered by the cursor.
-                                    if (!Globals.Database.NpcOverheadInfo && isNpc)
-                                    {
-                                        en.Value.DrawName(null);
-                                    }
+                if (en.Value.IsStealthed && !isPartyMate || !en.Value.WorldPos.Contains(mousePos.X, mousePos.Y))
+                {
+                    en.Value.IsHovered = false;
+                    continue;
+                }
 
-                                    // If PlayerOverheadInfo is toggled off, draw Players
-                                    // overhead information only when hovered by the cursor.
-                                    if (!Globals.Database.PlayerOverheadInfo && isOtherPlayer &&
-                                        !isFriend && !isGuildMate && !isPartyMate)
-                                    {
-                                        en.Value.DrawName(null);
-                                    }
+                // Consider the entity at this iteration hovered by the cursor.
+                en.Value.IsHovered = true;
 
-                                    // If PartyMemberOverheadInfo is toggled off, draw Party Members
-                                    // overhead information only when hovered by the cursor.
-                                    if (!Globals.Database.PartyMemberOverheadInfo && isPartyMate)
-                                    {
-                                        en.Value.DrawName(null);
-                                    }
+                // If MyOverheadInfo is toggled off, draw local player's overhead information.
+                if (!Globals.Database.MyOverheadInfo && isMe)
+                {
+                    en.Value.DrawName(null);
+                }
 
-                                    // If FriendOverheadInfo & GuildMemberOverheadInfo are off,
-                                    // let's prevent double draw / overlapping.
-                                    if (!Globals.Database.FriendOverheadInfo && isFriend &&
-                                        !Globals.Database.GuildMemberOverheadInfo && isGuildMate && !isPartyMate)
-                                    {
-                                        en.Value.DrawName(null);
+                // If NpcOverheadInfo is toggled off, draw NPCs overhead information.
+                else if (!Globals.Database.NpcOverheadInfo && isNpc)
+                {
+                    en.Value.DrawName(null);
+                }
 
-                                        continue;
-                                    }
+                // If PlayerOverheadInfo is toggled off, draw other player's overhead information.
+                else if (!Globals.Database.PlayerOverheadInfo && isOtherPlayer &&
```

Comment:
In particular this if statement looks like it might be an else condition, but I am not sure because the if-blocks below it are too tangled and I can't tell just by eyeballing it.